packages/gitbook/src/components/primitives/KeyboardShortcut.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import { type ClassValue, tcls } from '@/lib/tailwind';
+import * as React from 'react';
+
+export function KeyboardShortcut(props: { keys: string[]; className?: ClassValue }) {
+    const { keys, className } = props;
+
+    const [operatingSystem, setOperatingSystem] = React.useState<string | null>(null);
+
+    React.useEffect(() => {
+        function getOperatingSystem() {
+            const platform = navigator.platform.toLowerCase();
+
+            if (platform.includes('mac')) return 'mac';
+            if (platform.includes('win')) return 'win';
+
+            return 'win';
+        }
```

Comment:
```suggestion
function getOperatingSystem() {
    const platform = navigator.platform.toLowerCase();
    if (platform.includes('mac')) return 'mac';
    return 'win';
}
```

Put this function outside the react component

and do this : 

```javascript
React.useEffect(() => {
    setOperatingSystem(getOperatingSystem());
}, []);
```